discovery-provider/src/models.py

```diff
@@ -13,8 +18,59 @@
     Enum,
     PrimaryKeyConstraint,
 )
+from src.model_validator import ModelValidator
 
 Base = declarative_base()
+logger = logging.getLogger(__name__)
+
+def validate_field_helper(field, value, model):
+    # TODO: need to write custom validator for these datetime fields as jsonschema
+    # validates datetime in format 2018-11-13T20:20:39+00:00, not a format we use
+    # also not totally necessary as these fields are created server side
+    if field in ('created_at', 'updated_at'):
+        return value
+
+    to_validate = {
+        field: value
+    }
+    try:
+        ModelValidator.validate(to_validate=to_validate, model=model, field=field)
+    except ValidationError as e:
+        value = get_default_value(field, value, model, e)
+    except BaseException as e:
+        logger.error(f"Validation failed: {e}")
+
+    return value
+
+def get_default_value(field, value, model, e):
+    field_props = ModelValidator.get_properties_for_field(model, field)
+    schema_type_field = field_props['type'] # type field from the schema. this can either be a string or list
+    default_value = field_props['default']
+
+    # If the schema indicates this field is equal to object(if string) or contains object(if list) and
+    # the default value isn't set in the schema, set to SQL null, otherwise JSONB columns get
+    # set to string 'null'.
+    # Other fields can be set to their regular defaults or None.
+    if not default_value:
+        # if schema_type_field is defined as a list, need to check if 'object' is in list, else check string
+        if isinstance(schema_type_field, list) and 'object' in schema_type_field:
+            default_value = null() # sql null
+        elif schema_type_field == 'object':
+            default_value = null() # sql null
+
```

Comment:
is it possible for an else case to happen here? or is it ok that default_value gets returned out as falsey/None?

Reply:
its okay that default value is None. we just need to special case JSON to sql null which is what these two cases are